inc/saihostintf.h

```diff
@@ -230,6 +230,18 @@ typedef enum _sai_hostif_trap_id_t
     /** default packet action is forward */
     SAI_HOSTIF_TRAP_ID_MLD_V2_REPORT = 0x0000200f,
 
+    /** IP packets to local router IP address (routes with SAI_ROUTE_ATTR_NEXT_HOP_ID=SAI_SWITCH_ATTR_CPU_PORT)
```

Comment:
Hi,
The motivation for having specific traps is to be able to control the Qos settings – cpu queue and policer rate limit, on these specific types of traffic.
There is actual use case for this in Sonic.
We don't expect deep packet inspection, basically just match ip is ip2me, and the standard ports for the protocol.


From: Atit Jain [mailto:notifications@github.com]
Sent: Tuesday, May 17, 2016 10:01 AM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Itai Baz <itaib@mellanox.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] Add new traps : IP2ME, SSH and SNMP (#105)


In inc/saihostintf.h<https://github.com/opencomputeproject/SAI/pull/105#discussion_r63472582>:

> @@ -230,6 +230,18 @@ typedef enum _sai_hostif_trap_id_t

>      /** default packet action is forward */

>      SAI_HOSTIF_TRAP_ID_MLD_V2_REPORT = 0x0000200f,

>

> +    /** IP packets to local router IP address (routes with SAI_ROUTE_ATTR_NEXT_HOP_ID=SAI_SWITCH_ATTR_CPU_PORT)

  1.  My IP should be hit in the route/host table and then should go the CPU as pkt cmd. There is no other possible logical action. Worst case some user might want to drop.
  2.  Why we need specific TRAP IDs. Also do we expect the HW to do deeper packet inspection to even figure out application.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/opencomputeproject/SAI/pull/105/files/f346e1faf895e7c2127d66df5b0dc0a24229a843#r63472582>